### PR TITLE
POL-125 Scheduled Report With Estimates

### PR DESCRIPTION
**Pull request by LegacyAcres**
### Description

Adds 3 months of estimated cost to scheduled report

### Issues Resolved

N/A

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

**Comment by cdwilhelm** [comment]
@LegacyAcres  please rename the Pull Request to include the Jira project key and name of policy.  Thx.

File: cost/scheduled_reports_with_estimates/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+Scheduled Report
```
Comment (cdwilhelm) [inline cost/scheduled_reports_with_estimates/README.md]: Looks like this readme was copied from the schedule report.  Please update to match the policy, and is needs to be in Markdown format.  See the README_GUIDELINES

File: cost/scheduled_reports_with_estimates/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Change log
```
Comment (cdwilhelm) [inline cost/scheduled_reports_with_estimates/CHANGELOG.md]: Looks like this was copied from the Schedule Report Policy.  It should have only v1.0

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt [hunk 3]
```diff
@@ -0,0 +1,1005 @@
...
+Total cost during previous full week (Monday-Sunday) and previous full month.
+Total cost during current (incomplete) week and month.
+
+We recommend running this policy on a weekly cadence and applying it to your master account.
+
+_Note 1: The last 3 days of data in the current week or month will contain incomplete data._
+_Note 2: The account you apply the policy to is unimportant as Optima metrics are scoped to the Org._
+
+See [README](https://github.com/rightscale/policy_templates/tree/master/cost/scheduled_reports) for more details"
+long_description "Version: 1.6"
```
Comment (cdwilhelm) [inline cost/scheduled_reports_with_estimates/costs_forecasting.pt]: should be version 1.0

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt [hunk 4]
```diff
@@ -0,0 +1,1005 @@
...
+Daily average cost across the last week and last month.
+Total cost during previous full week (Monday-Sunday) and previous full month.
+Total cost during current (incomplete) week and month.
+
+We recommend running this policy on a weekly cadence and applying it to your master account.
+
+_Note 1: The last 3 days of data in the current week or month will contain incomplete data._
+_Note 2: The account you apply the policy to is unimportant as Optima metrics are scoped to the Org._
+
+See [README](https://github.com/rightscale/policy_templates/tree/master/cost/scheduled_reports) for more details"
```
Comment (cdwilhelm) [inline cost/scheduled_reports_with_estimates/costs_forecasting.pt]: Link to readme is wrong

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt [hunk 5]
```diff
@@ -0,0 +1,1005 @@
...
+end
+escalation "escalation_send_email" do
+  email $param_email
+end
+
+policy "policy_scheduled_report" do
+  validate $ds_generated_report_with_estimates do
+    summary_template "{{ rs_org_name }} (Org ID: {{ rs_org_id }}): Full Cost Scheduled Report"
+    detail_template <<-EOS
+# Full Cost Report for {{ rs_org_name }} - {{ data.currentMonthName }}
```
[the remark this answers is not in the document]
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/costs_forecasting.pt]: It was what the scheduled reports policy called it... open to suggestions

File: cost/scheduled_reports_with_estimates/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/README.md]: done

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt [hunk 6]
```diff
@@ -0,0 +1,1005 @@
+name "Scheduled Report With Estimates"
+rs_pt_ver 20180301
+type "policy"
+short_description "This policy allows you to set up scheduled reports that will provide summaries of cloud cost across all resources in the billing centers you specify, delivered to any email addresses you specify. The policy will report the following:
```
[the remark this answers is not in the document]
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/costs_forecasting.pt]: done

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt [hunk 7]
```diff
@@ -0,0 +1,1005 @@
...
+
+_Note 1: The last 3 days of data in the current week or month will contain incomplete data._
+_Note 2: The account you apply the policy to is unimportant as Optima metrics are scoped to the Org._
+
+See [README](https://github.com/rightscale/policy_templates/tree/master/cost/scheduled_reports) for more details"
+long_description "Version: 1.6"
+severity "low"
+category "Cost"
+
+## commented out until the optima resource is added to policies.
```
[the remark this answers is not in the document]
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/costs_forecasting.pt]: done

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt [hunk 8]
```diff
@@ -0,0 +1,1005 @@
...
+
+|Range|Daily Average|Previous|Current (incomplete)|
+|:---:|------------:|-------:|-------------------:|
+{{ range data.reportData -}}
+  | {{ .range }} | {{ .avg }} | {{ .previous}} | {{ .current }} |
+{{ end -}}
+
+The 3 months after the current month of {{ data.currentMonthName }} are estimates based on past cost trends.
+
+For more information on this report, please view the [README](https://github.com/rightscale/policy_templates/tree/master/cost/scheduled_reports).
```
[the remark this answers is not in the document]
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/costs_forecasting.pt]: done

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt [hunk 9]
```diff
@@ -0,0 +1,1005 @@
...
+
+  EOS
+end
+escalation "escalation_send_email" do
+  email $param_email
+end
+
+policy "policy_scheduled_report" do
+  validate $ds_generated_report_with_estimates do
+    summary_template "{{ rs_org_name }} (Org ID: {{ rs_org_id }}): Full Cost Scheduled Report"
```
[the remark this answers is not in the document]
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/costs_forecasting.pt]: It was what the scheduled reports policy called it... open to suggestions

File: cost/scheduled_reports_with_estimates/README.md (hunk 1 above)
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/README.md]: done

File: cost/scheduled_reports_with_estimates/CHANGELOG.md (hunk 2 above)
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/CHANGELOG.md]: done

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt (hunk 3 above)
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/costs_forecasting.pt]: done

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt (hunk 4 above)
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/costs_forecasting.pt]: done

File: cost/scheduled_reports_with_estimates/CHANGELOG.md (hunk 2 above)
Comment (cdwilhelm) [reply cost/scheduled_reports_with_estimates/CHANGELOG.md]: I don't see your changes.  You need to remove all versions except v1.0

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt (hunk 9 above)
[the remark this answers is not in the document]
Comment (cdwilhelm) [reply cost/scheduled_reports_with_estimates/costs_forecasting.pt]: This isn't the scheduled report policy, this is a new policy so it needs new details.  Maybe Estimated Cost, maybe talk among the team to get more details.

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt (hunk 5 above)
[the remark this answers is not in the document]
Comment (cdwilhelm) [reply cost/scheduled_reports_with_estimates/costs_forecasting.pt]: same response as above

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt (hunk 9 above)
[the remark this answers is not in the document]
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/costs_forecasting.pt]: It basically is the scheduled report policy though the only addition was the addition of the estimated costs

File: cost/scheduled_reports_with_estimates/README.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+Current - Weekly: Total costs during current (incomplete) week.
+Current - Monthly: Total costs during current (incomplete) month.
+Estimated Cost for each metrics for the next 3 months based on previous costs.
+
+We recommend running this policy on a weekly cadence and applying it to your master account.
+
+Note 1: The last 3 days of data in the current week or month will contain incomplete data.
+Note 2: The account you apply the policy to is unimportant as Optima metrics are scoped to the Org.
+
+Cost Metrics
```
Comment (cdwilhelm) [inline cost/scheduled_reports_with_estimates/README.md]: needs heading

File: cost/scheduled_reports_with_estimates/README.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
+# Scheduled Report with Estimates
+
+## What it does
+
+This policy allows you to set up scheduled reports that will provide summaries of cloud cost across all resources in the billing centers you specify, delivered to any email addresses you specify. The policy will report the following:
```
Comment (cdwilhelm) [inline cost/scheduled_reports_with_estimates/README.md]: should reflect details about 3 months estimate

File: cost/scheduled_reports_with_estimates/README.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+We recommend running this policy on a weekly cadence and applying it to your master account.
+
+Note 1: The last 3 days of data in the current week or month will contain incomplete data.
+Note 2: The account you apply the policy to is unimportant as Optima metrics are scoped to the Org.
+
+Cost Metrics
+There are four cost metrics to choose from.
+
+Unamortized Unblended - One-time and upfront costs are shown at the time of purchase. (AWS Only) Savings from reserved instances are applied first to matching instances in the account where it was purchased.
```
Comment (cdwilhelm) [inline cost/scheduled_reports_with_estimates/README.md]: these should have bullets.

File: cost/scheduled_reports_with_estimates/README.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+Amortized Unblended - One-time and upfront costs are spread evenly over the term of the item purchased. (AWS Only) Savings from reserved instances are applied first to matching instances in the account where it was purchased.
+Unamortized Blended - One-time and upfront costs are shown at the time of purchase. (AWS Only) Saving from reserved instances are shared equally by all matching instances in all accounts.
+Amortized Blended - One-time and upfront costs are spread evenly over the term of the item purchased. (AWS Only) Saving from reserved instances are shared equally by all matching instances in all accounts.
+Input Parameters
```
Comment (cdwilhelm) [inline cost/scheduled_reports_with_estimates/README.md]: This section is out of place and seems to be duplcated as it also exists below.

File: cost/scheduled_reports_with_estimates/README.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+This policy has the following input parameters required when launching the policy.
+
+## Functional Details
+
+The policy leverages the RightScale APis to report on billing data and send an email report
+
+### Input Parameters
+
+-Email list - Email addresses of the recipients you wish to notify
```
Comment (cdwilhelm) [inline cost/scheduled_reports_with_estimates/README.md]: these are not bulleted, add a space after -

File: cost/scheduled_reports_with_estimates/README.md [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+
+
+### Required RightScale Roles
+
+This policy requires permissions to access RightScale resources (Optima). Before applying this policy add the following roles to the user applying the policy. The roles should be applied to all Accounts where the policy will run or the Organization. For more information on modifying roles visit the Governance Docs
+Optima - billing_center_viewer
+
+### Supported Clouds
+
+AWS
```
Comment (cdwilhelm) [inline cost/scheduled_reports_with_estimates/README.md]: each cloud should be bulleted.

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt [hunk 16]
```diff
@@ -0,0 +1,998 @@
...
+We recommend running this policy on a weekly cadence and applying it to your master account.
+
+_Note 1: The last 3 days of data in the current week or month will contain incomplete data._
+_Note 2: The account you apply the policy to is unimportant as Optima metrics are scoped to the Org._
+
+See [README](https://github.com/rightscale/policy_templates/tree/master/cost/scheduled_reports_with_estimates) for more details"
+long_description "Version: 1.0"
+severity "low"
+category "Cost"
+
```
Comment (cdwilhelm) [inline cost/scheduled_reports_with_estimates/costs_forecasting.pt]: add tenancy "single" for org level policies.

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt (hunk 9 above)
[the remark this answers is not in the document]
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/costs_forecasting.pt]: added estimated cost

File: cost/scheduled_reports_with_estimates/costs_forecasting.pt (hunk 5 above)
[the remark this answers is not in the document]
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/costs_forecasting.pt]: added estimated cost

File: cost/scheduled_reports_with_estimates/README.md (hunk 10 above)
Comment (cdwilhelm) [reply cost/scheduled_reports_with_estimates/README.md]: To add a heading you need to add ## in front of Cost Metrics

File: cost/scheduled_reports_with_estimates/README.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+- Email list - Email addresses of the recipients you wish to notify
+- Billing Center List - List of top level Billing Center names you want to report on. Names must be exactly as shown in Optima. Leave the field blank to report on all top level Billing Centers.
+- Cost Metric - See Cost Metrics above for details on selection.
+- Graph Dimension - The cost dimension to break out the cost data in the embedded bar chart image
+
+
+### Required RightScale Roles
+
+This policy requires permissions to access RightScale resources (Optima). Before applying this policy add the following roles to the user applying the policy. The roles should be applied to all Accounts where the policy will run or the Organization. For more information on modifying roles visit the Governance Docs
+Optima - billing_center_viewer
```
Comment (cdwilhelm) [inline cost/scheduled_reports_with_estimates/README.md]: Add a new line before the roles and make it a bullet.

File: cost/scheduled_reports_with_estimates/README.md (hunk 10 above)
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/README.md]: done

File: cost/scheduled_reports_with_estimates/README.md (hunk 17 above)
Comment (LegacyAcres) [reply cost/scheduled_reports_with_estimates/README.md]: done

**Comment by cdwilhelm** [comment]
@LegacyAcres   if this is ready to merge, please remove the Do No Merge label and feel free to merge.

**Pull request merged**